src/jl_uv.c

```diff
@@ -681,11 +602,20 @@ int jl_printf(uv_stream_t *s, const char *format, ...)
     return c;
 }
 
-char *jl_bufptr(ios_t *s)
+DLLEXPORT void jl_safe_printf(const char *fmt, ...)
 {
-    return s->buf;
+    static char buf[1000];
+
+    va_list args;
+    va_start(args, fmt);
+    // Not async signal safe on some platforms?
+    vsnprintf(buf, sizeof(buf), fmt, args);
+    va_end(args);
+
+    write(STDERR_FILENO, buf, strlen(buf));
```

Comment:
i don't see any such statement: http://linux.die.net/man/3/vsnprintf. it seems that guarantee may be a BSD feature.

microsoft explicitly doesn't include it: https://msdn.microsoft.com/en-us/library/1kt27hek.aspx